packages/react-aria-components/src/GridList.tsx

```diff
@@ -402,10 +402,25 @@ function RootDropIndicator() {
   }
 
   return (
-    <div role="row" aria-hidden={dropIndicatorProps['aria-hidden']}>
+    <div role="row" aria-hidden={dropIndicatorProps['aria-hidden']} ref={fixGap}>
       <div role="gridcell">
         <div role="button" {...visuallyHiddenProps} {...dropIndicatorProps} ref={ref} />
       </div>
     </div>
   );
 }
+
+// Applies a negative margin to offset the gap of a parent flex or grid layout.
+function fixGap(element: HTMLElement | null) {
```

Comment:
Is this a bad idea? 😂 

Ideally the developer wouldn't even know about these hidden elements. They are zero height but if you put them in a flex layout with gap they still have an effect unfortunately.

Reply:
This seems bad, I'd prefer to explicitly do it to our examples and point it out.
There are still ways to break this by setting position absolute on them, in which case, you don't need the negative margins. Users may use flex/grid to display the actual rows and measure where these appear on demand.

Reply:
Yeah but these elements aren't provided by the user, so they won't be able to style them easily.

Actually making them position absolute might not be a bad idea... 🤔

Reply:
in some regards easier, won't affect layout, in other regards, harder, width/height 100% are more difficult

anyway we could expose these as elements that the user could provide? I've given it no thought, so sorry if that's a dumb question

Reply:
I don't think the user needs to know about them. They are never visible, just visually hidden elements for screen readers, so there not really a reason to style them other than this. So if we can make it work automatically that's better. Looks like position: absolute works well.